packages/lit-html/src/lit-html.ts

```diff
@@ -466,13 +466,38 @@ const COMMENT_PART = 7;
  * [Rendering](https://lit.dev/docs/components/rendering) for more information.
  *
  */
-export type TemplateResult<T extends ResultType = ResultType> = {
+export type UncompiledTemplateResult<T extends ResultType = ResultType> = {
   // This property needs to remain unminified.
   ['_$litType$']: T;
   strings: TemplateStringsArray;
   values: unknown[];
 };
 
+export type MaybeCompiledTemplateResult<T extends ResultType = ResultType> =
```

Comment:
Nit: This could use a JSDoc to clarify why this is used. Probably want to call out that this should ideally be the type of `html` or `svg` when you're trying to target both uncompiled and compiled environments.

Reply:
Good point! Done